tests/MiddlewareTest.php

```diff
@@ -28,7 +28,7 @@ class MiddlewareTest extends TestCase
 {
     public function set_up()
     {
-        \GuzzleHttp\Promise\queue()->run();
+        \GuzzleHttp\Promise\Utils::queue()->run();
```

Comment:
This fix should be moved into master, already, since `GuzzleHttp\Promise\queue` is deprecated on 1.x, too.

Reply:
Thank you Graham, we merged this today in #2711; will address the other comments soon